### PR TITLE
wasm-decompile: documentation.

### PR DESCRIPTION
**Pull request by aardappel**
Added a language description, a man page, and some links in the
general readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ WABT (we pronounce it "wabbit") is a suite of tools for WebAssembly, including:
  - [**wasm2wat**](https://webassembly.github.io/wabt/doc/wasm2wat.1.html): the inverse of wat2wasm, translate from the binary format back to the text format (also known as a .wat)
  - [**wasm-objdump**](https://webassembly.github.io/wabt/doc/wasm-objdump.1.html): print information about a wasm binary. Similiar to objdump.
  - [**wasm-interp**](https://webassembly.github.io/wabt/doc/wasm-interp.1.html): decode and run a WebAssembly binary file using a stack-based interpreter
+ - [**wasm-decompile**](https://webassembly.github.io/wabt/doc/wasm-decompile.1.html): decompile a wasm binary into readable C-like syntax.
```
Comment (sbc100) [inline README.md]: Looks like these html files are stored in the `gh-pages` branch.     I guess we should have some way to automatically update them?  But I know nothing of this..

File: README.md [hunk 2]
```diff
@@ -259,6 +260,24 @@ You can use `--help` to get additional help:
 $ out/wasm-interp --help
 ```
 
+## Running wasm-decompile
+
+For example:
+
+```sh
+# parse binary file test.wasm and write text file test.dcmp
+$ out/wasm-decompile test.wasm -o test.dcmp
```
[the remark this answers is not in the document]
Comment (aardappel) [reply README.md]: I'll do all.

File: docs/decompiler.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+Though compiling this output code back into a wasm module is possible,
+such functionality is currently not provided. The format is very low-level,
+much like Wasm itself, so even though it looks more high level than the .wat
+format, it wouldn't be any more suitable for general purpose programming.
+
+## Language guide.
+
+This section shows some aspects of the language in terms of how they map to
+Wasm and/or how they might differ from a typical C-like language. It does
+not try to define the actual semantics of Wasm, the reader is expected to
```
[the remark this answers is not in the document]
Comment (aardappel) [reply docs/decompiler.md]: According to https://webassembly.org/: `WebAssembly (abbreviated Wasm)` (and then goes on to use `Wasm` many times more on the page).

File: README.md (hunk 1 above)
Comment (aardappel) [reply README.md]: yeah not sure how this is supposed to work.. I guess @binji knows.

File: docs/decompiler.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbc100) [reply docs/decompiler.md]: I stand corrected!

File: README.md (hunk 1 above)
Comment (binji) [reply README.md]: Nope, I think it's manual for now. Would be nice to make it automatic though.

Comment (Horcrux7) [reply README.md]: Github supports to use a sub directory of master branch instead of gh-pages branch now. Then you need not to sync it.

Comment (sbc100) [reply README.md]: Sweet. That sounds way better.

Comment (aardappel) [reply README.md]: Yes, in fact for `docs/` it can be fully automatic, that's why I put this file there.

**Pull request merged**